app/main.py

```diff
@@ -1,25 +1,88 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: Any = None) -> None:
+        getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if self.min_amount > value or self.max_amount < value:
```

Comment:
You could consider writing this condition without `or`, e.g. `if not self.min_amount <= value <= self.max_amount:`